src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -231,6 +231,26 @@ public async Task<CodeFixCollection> GetDocumentFixAllForIdInSpan(Document docum
             return result.ToImmutableAndFree().FirstOrDefault();
         }
 
+        public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document document, string diagnosticId, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var textSpan = new TextSpan(0, tree.Length);
+
+            var fixCollection = await GetDocumentFixAllForIdInSpanAsync(
+                document, textSpan, diagnosticId, cancellationToken).ConfigureAwait(false);
```

Comment:
is this ever called with anything but the full span of hte doucment?  I couldn't find thsi function in roslyn today.  If that's the case, having the caller have to figure out hte full span, just to pass this in seems strange.  why not do it inside the function itself?

Reply:
Prior to this pull request, the method is missing the Async suffix. I'll check callers tomorrow when I'm updating the pull request per the feedback from this review so I can better answer the question.

Reply:
great!

Reply:
It looks like the answer is 'no', but it also wasn't clear that this pull request would benefit from changing it.